contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+        _burnOethbOnTheContract();
+    }
+
+    function _withdraw(address _recipient, uint256 _amount) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Collect the AERO token from the gauge
+     */
+    function _collectRewardTokens() internal override {
+        clGauge.getReward(tokenId);
+        super._collectRewardTokens();
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     * @param _asset Address of the asset to approve
+     * @param _pToken Address of the pToken
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(_asset).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /**
+     * @dev Approve the spending of all assets
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
+
+        // we could in theory deposit to the strategy and forget to call rebalance in the same
+        // governance transaction batch. In that case the WETH that is on the strategy contract
+        // also needs to be accounted for.
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // gotten burned yet.
+        uint256 oethbBalance = IERC20(OETHb).balanceOf(address(this));
```

Comment:
Anyone can send OETHb to the contract to make this function return a higher value. However, I don't see any reason how that'd make the strategy vulnerable

Reply:
Yup agreed. They would in effect be giving up their tokens for the higher yield in the strategy. Since the strategy would either deposit or burn the extra OETHb tokens resulting in a better rebase